src/main/java/Storage.java

```diff
@@ -0,0 +1,131 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    protected String filePath;
+
+    /**
+     * Manages file management.
+     * Will load, update, and append the associated .txt file.
+     * @param filePath refers to the path of the .txt file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the .txt file into the program.
+     * @return ArrayList of Task specified in the .txt file
+     * @throws IOException when unable to resolve date
+     */
+    public ArrayList<Task> load() throws IOException {
+        File file = new File(filePath);
+        ArrayList<Task> list = new ArrayList<>();
+        Scanner sc = new Scanner(file);
+        while (sc.hasNextLine()) {
+            String[] task = sc.nextLine().replaceAll(", ", ",").split(",");
+
+            if (task[0].equals("T")) {
+                Task tsk = new Todo(task[2]);
+                if (task[1].equals("1")) {
+                    tsk.markAsDone();
+                }
+                list.add(tsk);
+            } else if (task[0].equals("D")) {
+                try {
+                    SimpleDateFormat formatter = new SimpleDateFormat("EEEEE MMMMM dd HH:mm:ss z yyyy");
+                    Task tsk = new Deadline(task[2], formatter.parse(task[3]));
+                    if (task[1].equals("1")) {
+                        tsk.markAsDone();
+                    }
+                    list.add(tsk);
+                } catch (java.text.ParseException exp) {
+                    exp.printStackTrace();
+                    break;
+                }
+            } else {
```

Comment:
A extended if-elseif-else outer loop with a try-catch inner loop here. Perhaps can extract the inner loop into another function to enhance readability. But code is already clear with reference to JavaDoc written.